app/helpers/settings_helper.rb

```diff
@@ -40,4 +40,33 @@ def translate_full_name(setting)
     fullname = setting.full_name.nil? ? setting.name : _(setting.full_name)
     trunc_with_tooltip(fullname, 32, setting.name, false)
   end
+
+  def grouped_settings(settings)
+    settings.each_with_object({}) do |setting, memo|
+      hash = setting_to_hash(setting)
+      if memo[setting.category]
+        memo[setting.category] << hash
+      else
+        memo[setting.category] = [hash]
+      end
+      memo
+    end
+  end
+
+  def setting_to_hash(setting)
+    {
+      :id => setting.id,
+      :name => setting.name,
+      :category => setting.category,
+      :description => setting.description,
+      :settings_type => setting.settings_type,
+      :default => setting.default,
+      :readonly => setting.readonly?,
+      :full_name => setting.full_name,
```

Comment:
The `full_name` is not being translated anywhere else IIUC.
```suggestion
      :full_name => setting.full_name.nil? ? setting.name : _(setting.full_name),
```